### ISSUE TITLE
PhoneNumberKit fails to build when using Xcode 13.2 beta 2

### ISSUE THREAD

**Issue by nikolaykasyanov** [marmelroy/PhoneNumberKit#489]
I believe it to be a compiler regression which I reported to Apple as FB9767233, although an issue here is still useful I believe.

Steps to reproduce:

1.` git clone git@github.com:marmelroy/PhoneNumberKit.git`
2. `cd PhoneNumberKit`
3. Enable the experimental build system feature: `defaults write com.apple.dt.XCBuild EnableSwiftBuildSystemIntegration 1`
3. Edit `build.sh` to use an available iOS simulator version
4. `env DEVELOPER_DIR=/Applications/Xcode-13.2.beta.2.app/Contents/Developer ./build.sh build`

Expected result:
Build successfully completes.

Actual result:
```
PhoneNumberTextField.swift:124:62: error: 'didPressFlagButton()' is only available in application extensions for iOS 11.0 or newer
                flagButton.addTarget(self, action: #selector(didPressFlagButton), for: .touchUpInside)
                                                             ^

...

Testing failed:
	'didPressFlagButton()' is only available in application extensions for iOS 11.0 or newer
	Testing cancelled because the build failed.
```

**Update**: that’s only the case when this new experimental setting is enabled:

```bash
defaults write com.apple.dt.XCBuild EnableSwiftBuildSystemIntegration 1
```

I updated the repro steps accordingly.

**Comment by nikolaykasyanov** [comment]
Workarounds: https://forums.swift.org/t/xcode-13-2b2-swift-build-system-integration-fails-to-recognise-availability-checks/53538/3

**Comment by nikolaykasyanov** [comment]
Fixed in Xcode 13.2 RC.
